apps/clamav.sh

```diff
@@ -0,0 +1,186 @@
+#!/bin/bash
+
+# T&M Hansson IT AB © - 2020, https://www.hanssonit.se/
+
+# shellcheck disable=2034,2059
+true
+SCRIPT_NAME="ClamAV"
+SCRIPT_EXPLAINER="This script installs the open-source antivirus-software ClamAV on your server \
+and configures Nextcloud to detect infected files already during the upload.
+At the end of the script, you will be able to choose to setup a weekly full scan of all files."
+# shellcheck source=lib.sh
+. <(curl -sL https://raw.githubusercontent.com/nextcloud/vm/master/lib.sh)
+
+# Check for errors + debug code and abort if something isn't right
+# 1 = ON
+# 0 = OFF
+DEBUG=0
+debug_mode
+
+# Check if root
+root_check
+
+# Show explainer
+explainer_popup
+
+# Needs 1 GB alone
+ram_check 3 "ClamAV"
+cpu_check 2 "ClamAV"
+
+# Check if webmin is already installed
+print_text_in_color "$ICyan" "Checking if ClamAV is already installed..."
+if is_this_installed clamav-daemon || is_this_installed clamav || is_this_installed clamav-freshclam
+then
+    # Ask for removal or reinstallation
+    reinstall_remove_menu
+    # Removal
+    apt purge clamav-daemon -y
+    apt purge clamav-freshclam -y
+    apt purge clamav -y
+    apt autoremove -y
+    rm -f /etc/systemd/system/clamav-daemon.service
+    rm -f "$SCRIPTS"/clamav-fullscan.sh
+    rm -f "$VMLOGS"/clamav-fullscan.log
+    crontab -u root -l | grep -v 'clamav-fullscan.sh'  | crontab -u root -
+    if is_app_installed files_antivirus
+    then
+        occ_command_no_check app:remove files_antivirus
+    fi
+    # Show successful uninstall if applicable
+    removal_popup
+else
+    print_text_in_color "$ICyan" "Installing ClamAV..."
+fi
+
+# Install needed tools
+apt update -q4 & spinner_loading
+apt install clamav clamav-freshclam clamav-daemon -y
+
+# stop freshclam and update the database
+check_command systemctl stop clamav-freshclam
+check_command freshclam
+check_command systemctl start clamav-freshclam
+
+# Edit ClamAV settings to fit the installation
+sed -i "s|^MaxDirectoryRecursion.*|MaxDirectoryRecursion 30|" /etc/clamav/clamd.conf
+sed -i "s|^MaxFileSize.*|MaxFileSize 100M|" /etc/clamav/clamd.conf
+sed -i "s|^PCREMaxFileSize.*|PCREMaxFileSize 100M|" /etc/clamav/clamd.conf
+sed -i "s|^StreamMaxLength.*|StreamMaxLength 100M|" /etc/clamav/clamd.conf
+
+# Start ClamAV
+check_command systemctl restart clamav-freshclam
+check_command systemctl restart clamav-daemon
+
```

Comment:
We should check for the existence of `/var/run/clamav/clamd.ctl` to be sure the daemon works properly - and then continue so that we don't clutter the system before we know everything is good.

Reply:
you mean something like this?
```
print_text_in_color "$ICyan" "Waiting for clamd to start up..."
while ! [ -f /var/run/clamav/clamd.ctl ]
do
    sleep 5
done
```

I mean we would need to have a check so that it doesnt run endless

Reply:
Maybe an until loop is better? 

Until like 10 retries or something.. and after than tell the user that something failed and report to $ISSUES.

Reply:
done